src/otx/algorithms/anomaly/adapters/anomalib/plugins/xpu_precision.py

```diff
@@ -0,0 +1,109 @@
+"""Plugin for mixed-precision training on XPU."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+
+from contextlib import contextmanager
+from typing import Any, Callable, Dict, Generator, Optional, Union
+
+import pytorch_lightning as pl
+import torch
+from lightning_fabric.utilities.types import Optimizable
+from pytorch_lightning.plugins.precision.precision_plugin import PrecisionPlugin
+from pytorch_lightning.utilities import GradClipAlgorithmType
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from torch import Tensor
+from torch.optim import LBFGS, Optimizer
+
+
+class MixedPrecisionXPUPlugin(PrecisionPlugin):
```

Comment:
As you know, current IPEX doesn't support fp16 training and bfp16 mixed precision doesn't need scaler. Is it for fp16 mixed precision training later?

Reply:
No, it's not for FP16. In theory, you're right, BFP16 doesn't need a scaler, because it fully covers the numerical range of FP32. Lightning implicitly disables scaler if BFP16 is set for CUDA.
In recent discussion @sungchul2 claimed what we might need a scaler for BFP16 as well, and also IPEX team has a scaler for BFP16 on CPU, so I decided to leave a room for it in current implementation.